stl/inc/yvals.h

```diff
@@ -164,10 +164,18 @@ _STL_DISABLE_CLANG_WARNINGS
 #error _ITERATOR_DEBUG_LEVEL != 0 must imply _CONTAINER_DEBUG_LEVEL == 1.
 #endif // _ITERATOR_DEBUG_LEVEL != 0 && _CONTAINER_DEBUG_LEVEL == 0
 
-#define _STL_REPORT_ERROR(mesg)              \
-    do {                                     \
-        _RPTF0(_CRT_ASSERT, mesg);           \
-        _CRT_SECURE_INVALID_PARAMETER(mesg); \
+#ifndef _STL_CRT_SECURE_INVALID_PARAMETER
+#ifdef _DEBUG
+#define _STL_CRT_SECURE_INVALID_PARAMETER(expr) ::_invalid_parameter(_CRT_WIDE(#expr), L"", __FILEW__, __LINE__, 0)

```

Comment:
This is great, especially your escape hatch design (allowing this macro to be user-overridden). I think this would benefit from a comment citing the issue this is solving, since otherwise this logic will look mysterious years down the road (and while `git blame` is powerful, it takes a while to perform such archaeology).